common/CMakeLists.txt

```diff
@@ -1,16 +1,79 @@
-# CMake configuration for common directory
+include_guard(GLOBAL)
 
-if(NOT DYNINST_OS_Windows)
-    include_directories(${PROJECT_SOURCE_DIR}/common/h ${PROJECT_SOURCE_DIR}/common/src)
-else()
-    include_directories(
-        ${PROJECT_SOURCE_DIR}/common/h # don't include common/src; anything from there can
-                                       # still collide with default includes.
-        # stupid Windows case-insensitive naming.
-        )
-endif()
+include(DyninstLibrary)
+
+configure_file(${PROJECT_SOURCE_DIR}/cmake/version.h.in
+               ${CMAKE_CURRENT_SOURCE_DIR}/h/dyninstversion.h)
```

Comment:
I put it in the source tree because I had explicitly listed it in the public headers for libcommon. I agree that having an immutable source tree is a good idea. Updated.

Reply:
Ah, no, I remember now. The generated file is assumed to be in the source tree because it's included by a whole bunch of other files. In master, it's explicitly listed as `configure_file(cmake/version.h.in common/h/dyninstversion.h)`. It's always been modifying the source tree.